### DISCUSSION TITLE
Help regarding WebSocket

### DISCUSSION THREAD

**Discussion by Rohit-2602** [microsoft/react-native-windows#12839] [General]
Hey, I'm currently exploring the implementation of WebSocket functionality within my React Native project. Here's a snippet of the WebSocket instantiation:
```
const ws = new WebSocket(websocketUrl, [], {
    headers: { ["Origin"]: origin },
});

or 

const ws = new WebSocket(websocketUrl, [], {
    headers: { Origin: origin },
});
```

This setup works fine on Android and iOS platforms. However, on the Windows platform, I've encountered an issue where the Origin header value gets overridden. I checked the native code and I found this piece of code -

// vnext/Shared/Networking/WinRTWebSocketResource.cpp
``` 

bool hasOriginHeader = false;
for (const auto &header : options) {
    m_socket.SetRequestHeader(header.first, winrt::to_hstring(header.second));
    if (boost::iequals(header.first, L"Origin")) {
      hasOriginHeader = true;
    }
}

if (!hasOriginHeader) {
      auto scheme = uri.SchemeName();
      auto host = uri.Host();
      auto port = uri.Port();

      if (scheme == L"ws") {
        scheme = L"http";
      } else if (scheme == L"wss") {
        scheme = L"https";
      }

      auto origin = winrt::hstring{scheme + L"://" + host + L":" + winrt::to_hstring(port)};
      m_socket.SetRequestHeader(L"Origin", std::move(origin));
}
```

In an attempt to address this, I manually attempted to hardcode the Origin value as follows (**updating the node_modules**)
(node_modules/react-native-windows/Shared/Networking/WinRTWebSocketResource.cpp)
```diff
- auto origin = winrt::hstring{scheme + L"://" + host + L":" + winrt::to_hstring(port)};
+ auto origin = winrt::hstring{ L"MyOrigin" };
```

This change resolves the issue in debug mode. However, it fails to function as expected in release mode.

I'm seeking assistance in resolving this issue/problem. Any insights or guidance on how to rectify this situation would be greatly appreciated.

**Comment by acoates-ms** [comment]
@JunielKatarn
